src/Microsoft.ApplicationInsights/Managed/Shared/Extensibility/Implementation/Tracing/HealthHeartbeatDefaultPayload.cs

```diff
@@ -0,0 +1,125 @@
+        {
+        }
+
+        public HealthHeartbeatDefaultPayload(IEnumerable<string> disableFields)
+        {
+            this.SetEnabledProperties(disableFields);
+        }
+
+        public IDictionary<string, HealthHeartbeatPropertyPayload> GetPayloadProperties()
+        {
+            var payload = new Dictionary<string, HealthHeartbeatPropertyPayload>();
+            foreach (string fieldName in this.enabledProperties)
+            {
+                switch (fieldName)
+                {
+                    case "runtimeFramework":
+                        payload.Add(fieldName, new HealthHeartbeatPropertyPayload()
+                        {
+                            IsHealthy = true,
+                            PayloadValue = this.GetRuntimeFrameworkVer()
+                        });
+                        break;
+                    case "targetFramework":
+                        payload.Add(fieldName, new HealthHeartbeatPropertyPayload()
+                        {
+                            IsHealthy = true,
+                            PayloadValue = this.GetTargetFrameworkVer()
+                        });
+                        break;
+                    case UpdatedFieldsPropertyKey:
+                        var updatedFieldItem = new HealthHeartbeatPropertyPayload()
+                        {
+                            IsHealthy = true,
+                            PayloadValue = string.Empty
+                        };
+                        updatedFieldItem.IsUpdated = false; // always set this to false
+                        payload.Add(fieldName, updatedFieldItem);
+                        break;
+                    default:
+                        payload.Add(fieldName, new HealthHeartbeatPropertyPayload()
+                        {
+                            IsHealthy = false,
+                            PayloadValue = "UNDEFINED"
+                        });
+                        break;
+                }
+            }
+
+            return payload;
+        }
+
+        private void SetEnabledProperties(IEnumerable<string> disabledFields)
+        {
+            if (disabledFields == null || disabledFields.Count() <= 0)
+            {
+                this.enabledProperties = DefaultFields.ToList();
+            }
+            else
+            {
+                this.enabledProperties = new List<string>();
+                foreach (string fieldName in DefaultFields)
+                {
+                    if (!disabledFields.Contains(fieldName, StringComparer.OrdinalIgnoreCase))
+                    {
+                        this.enabledProperties.Add(fieldName);
+                    }
+                }
+            }
+        }
+
+        private string GetTargetFrameworkVer()
+        {
+#if NET45
+            return "net45";
+#elif NET46
+            return "net46";
+#elif NETSTANDARD1_3
+            return "netstandard1.3";
+#else 
+            return "undefined";
```

Comment:
Consider `#error Unrecognized framework.` so that we don't forget to update it if we add a new target framework.